common-auth/src/main/java/com/hortonworks/registries/auth/KerberosLogin.java

```diff
@@ -246,6 +260,26 @@ private KerberosTicket getTGT() {
         return null;
     }
 
+    private void synchronizeReLogin() throws LoginException {
```

Comment:
I dont see any point having relogin and synchronizeReLogin. Please rename this as relogin and delete the other.

Reply:
The code in try {} block and in else {} block have same lines of code which has been abstracted out as reLogin(). Its better to keep this abstracted out, as synchronizedReLogin() is wrapper over reLogin() which hold the actual logic to reLogin() while synchornizedReLogin() simply provides a synchronization on top of it.